src/widgets/WidgetApi.ts

```diff
@@ -116,6 +127,10 @@ export class WidgetApi {
         return this.readyPromise;
     }
 
+    public addTerminateCallback(action) {
```

Comment:
actually, this function probably needs to change slightly. The `WidgetApi` class should extend `EventEmitter` (or whatever equivalent these days) and document something like `.on('terminate_requested')` being raised.